src/provider/index.ts

```diff
@@ -3,7 +3,7 @@ import {inject, injectable} from 'inversify';
 import {Identity} from 'openfin/_v2/main';
 import {ProviderIdentity} from 'openfin/_v2/api/interappbus/channel/channel';
 
-import {RaiseIntentPayload, APIFromClientTopic, OpenPayload, FindIntentPayload, FindIntentsByContextPayload, BroadcastPayload, APIFromClient, IntentListenerPayload, GetDesktopChannelsPayload, GetCurrentChannelPayload, ChannelGetMembersPayload, ChannelJoinPayload, ChannelTransport, DesktopChannelTransport, GetChannelByIdPayload, EventTransport, ChannelBroadcastPayload, ChannelGetCurrentContextPayload, ChannelAddContextListenerPayload, ChannelRemoveContextListenerPayload} from '../client/internal';
+import {RaiseIntentPayload, APIFromClientTopic, OpenPayload, FindIntentPayload, FindIntentsByContextPayload, BroadcastPayload, APIFromClient, IntentListenerPayload, GetDesktopChannelsPayload, GetCurrentChannelPayload, ChannelGetMembersPayload, ChannelJoinPayload, ChannelTransport, DesktopChannelTransport, GetChannelByIdPayload, EventTransport, ChannelBroadcastPayload, ChannelGetCurrentContextPayload, ChannelAddContextListenerPayload, ChannelRemoveContextListenerPayload, ChannelAddEventListenerPayload, ChannelRemoveEventListenerPayload} from '../client/internal';
```

Comment:
This is a lot of imports from one file. It was originally done as one because there was a very limited set of types that needed exporting. Probably a question for @pjbroadbent too, but do you think it would make sense to start splitting internal by functionality rather than having it all in one. Not sure where we would make the splits, but maybe worth discussion?

Reply:
Hmm, the first split I'd probably make would to pull out the API and Payload definitions (which themselves I'd be reluctant to split further), which wouldn't reduce the size of this import.

Reply:
Yea, as I said it's hard to see where there could be a logical separation. Lets have a quick chat offline and if we can't think of anything better then lets just leave it for now.

Reply:
Discussed - will leave as is